src/blueapi/cli/format.py

```diff
@@ -0,0 +1,126 @@
+import builtins
+import enum
+import json
+import sys
+import textwrap
+from functools import partial
+from pprint import pprint
+from textwrap import dedent, indent
+
+from blueapi.service.model import DeviceResponse, PlanResponse
+
+FALLBACK = pprint
+
+
+class OutputFormat(enum.Enum):
+    JSON = enum.auto()
+    FULL = enum.auto()
+    COMPACT = enum.auto()
+    PPRINT = enum.auto()
+
+    def display(self, obj, out=None):
+        out = out or sys.stdout
+        match self:
+            case OutputFormat.FULL:
+                display_full(obj, out)
+            case OutputFormat.COMPACT:
+                display_compact(obj, out)
+            case OutputFormat.JSON:
+                display_json(obj, out)
+            case OutputFormat.PPRINT:
+                display_pprint(obj, out)
+
+
+def display_full(obj, stream):
+    print = partial(builtins.print, file=stream)
+    match obj:
+        case PlanResponse(plans=plans):
+            for plan in plans:
+                print(plan.name)
+                if desc := plan.description:
+                    print(indent(dedent(desc).strip(), "    "))
+                if schema := plan.parameter_schema:
+                    print("    Schema")
+                    print(indent(json.dumps(schema, indent=2), "        "))
+        case DeviceResponse(devices=devices):
+            for dev in devices:
+                print(dev.name)
+                for proto in dev.protocols:
+                    print("    " + proto)
+        case other:
+            FALLBACK(other)
+
+
+def display_json(obj, stream):
+    print = partial(builtins.print, file=stream)
+    match obj:
+        case PlanResponse(plans=plans):
+            print(json.dumps([p.dict() for p in plans], indent=2))
+        case DeviceResponse(devices=devices):
+            print(json.dumps([d.dict() for d in devices], indent=2))
```

Comment:
Could: Just serialize anything from pydantic?

```suggestion
        case BaseModel():
            print(json.dumps(obj.dict(), indent=2))
```

Reply:
Will add it as a fallback but I wanted to avoid the `{"devices": [...] }` level of nesting in the json.